eng/native/configurecompiler.cmake

```diff
@@ -447,6 +447,15 @@ if (CLR_CMAKE_HOST_UNIX)
     add_compile_options(-Wno-incompatible-ms-struct)
 
     add_compile_options(-Wno-reserved-identifier)
+
+    # clang 16.0 introduced buffer hardening https://discourse.llvm.org/t/rfc-c-buffer-hardening/65734
+    # which we are not conforming to yet.
+    add_compile_options(-Wno-unsafe-buffer-usage)
+
+    # other clang 16.0 suppressions
+    add_compile_options(-Wno-single-bit-bitfield-constant-conversion)
+    add_compile_options(-Wno-cast-function-type-strict)
+    add_compile_options(-Wno-incompatible-function-pointer-types-strict)
```

Comment:
Do you know what needs this? I can't seem to find this needed in any other place. I also didn't see it on lld's man page.

Reply:
This came up when trying to build on Alpine Linux with clang 16. I reported it in https://github.com/dotnet/runtime/issues/81577 to not forget to investigate.